scripts/yetong02_dynamic_trajectory_planning.cpp

```diff
@@ -0,0 +1,280 @@
+/* ----------------------------------------------------------------------------
+ * GTDynamics Copyright 2020, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * See LICENSE for the license information
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file  dynamic_trajectory_planning.cpp
+ * @brief Dynamic trajectory planning problem of rotating up a cart-pole.
+ * @author Yetong Zhang
+ */
+
+#include <gtdynamics/dynamics/DynamicsGraph.h>
+#include <gtdynamics/factors/JointLimitFactor.h>
+#include <gtdynamics/factors/MinTorqueFactor.h>
+#include <gtdynamics/factors/PointGoalFactor.h>
+#include <gtdynamics/optimizer/ManifoldOptimizerType1.h>
+#include <gtdynamics/universal_robot/RobotModels.h>
+#include <gtdynamics/utils/Initializer.h>
+#include <gtsam/base/timing.h>
+#include <gtsam/nonlinear/LevenbergMarquardtOptimizer.h>
+#include <gtsam/slam/BetweenFactor.h>
+
+using namespace gtsam;
+using namespace gtdynamics;
+using gtsam::noiseModel::Isotropic;
+
+auto robot = CreateRobotFromFile(kUrdfPath + std::string("cart_pole.urdf"))
```

Comment:
Since this and the previous file are examples, the documentation should be made keeping pedagogy in mind. If someone new wants to use this, they should be able to see these examples, read the comments and fully understand how to use this machinery.

Please be as detailed as possible.

Reply:
Done